### ISSUE TITLE
Procfile parse not working after slug import

### ISSUE THREAD

**Issue by OwlyCode** [gliderlabs/herokuish#28]
Hello there,

I generated a slug using herokuish. I now try to run a process from it but fail to parse the Procfile. Here's what I do : 

```
FROM gliderlabs/herokuish

COPY bar.tgz /bar.tgz

RUN cat /bar.tgz | /bin/herokuish slug import

CMD /bin/herokuish procfile parse
```

The command exists with a 0 status code and nothing is displayed. The Procfile exists in `/app`. What could possibly go wrong ?

**Comment by mgood** [comment]
Try `herokuish procfile parse web`, or another argument. It only prints out the command defined in the Procfile for that key, so if you want to start the app, you probably want to use `herokuish procfile start` instead.

**Comment by OwlyCode** [comment]
I finally got what was wrong. I actually forgot to add the process name for the procfile parse. I then replaced it by a `procfile start web`. But still fail to have the slug running : 

```
bin/heroku-php-nginx: line 205: -dnewrelic.enabled=0: command not found
This program requires PHP 5.5.11 or newer; check your 'php' command.
```

**Comment by mgood** [comment]
Oh, you would need to run `herokuish buildpack build` (or the `/build` shortcut) as well before starting the app.

Have you taken a look at [Dokku](https://github.com/progrium/dokku) or [Deis](http://deis.io)? Both are platforms for running Heroku-style apps, and might be useful. Dokku uses Herokuish, and Deis will use it in the future.

If you're looking for a way to run Heroku-style apps on your own servers, both of those are good options (Dokku for small, single-server deployments, Deis for larger deployments). Or if you're implementing your own platform, you may want to look at how they're implemented.

**Comment by OwlyCode** [comment]
Thank you for the examples I'll give a look at the implementations. But I don't get why I need to do a build. I already ran the build and exported the slug. The one I am importing already have all the binaries.

**Comment by mgood** [comment]
Oh ok, sorry I missed that you had already done the build. There might be an issue with the upstream PHP buildpack version currently included by Herokuish.

**Comment by OwlyCode** [comment]
Looks to me that the files under `/app/.profile.d` are not sourced : 

```
root@c267d00a5a2a:/# /bin/herokuish procfile exec echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by mgood** [comment]
@OwlyCode in your command, the `$PATH` is not quoted, so it will be evaluated by your shell first, before running the `herokuish` command. To see the value of `PATH` within the herokuish environment you can quote it to make sure it's passed through before being evaluated:

    /bin/herokuish procfile exec echo '$PATH'

For example, compare:

    root@5ee5be635a22:/# /bin/herokuish procfile exec echo $PATH
    /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    root@5ee5be635a22:/# /bin/herokuish procfile exec echo '$PATH'
    /app/.heroku/python/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

It does look Herokuish doesn't include the most recent PHP buildpack yet, so you could try updating it to see if it's a buildpack issue that has been fixed in a more recent version. This should install the current newest release tag:

    herokuish buildpack install https://github.com/heroku/heroku-buildpack-php v67

**Comment by OwlyCode** [comment]
Hello ! Thank you for your help ! I finally solved this :

I executed `/bin/herokuish procfile exec echo '$PATH'` and you were right, the PATH was correctly set. So I tried to run the php binary directly with absolute path. Turns out the php binary was in "permission denied" (read, write and exec).

It was a Docker issue (https://github.com/docker/docker/issues/783).

Now everything is building and running correctly.

**Comment by mgood** [comment]
@OwlyCode ah, good. Thanks for continuing to investigate. What did you change to get it working for you? From the issue it sounds like some recent changes in AUFS and Docker have addressed it, so did you upgrade Docker, or find another solution?

**Comment by OwlyCode** [comment]
I upgraded Docker yes. And for the older versions I apply a workaround (moving the folder) as a temporary fix while working on other issues.
